great_expectations/datasource/fluent/interfaces.py

```diff
@@ -250,14 +288,17 @@ def add_batch_config(self, name: str) -> BatchConfig:
         assert isinstance(self.datasource, Datasource)
 
         batch_config = BatchConfig(name=name)
-        batch_config.set_data_asset(self)
-        self.__fields_set__.add("batch_configs")
-        if self.datasource.is_persisted():
-            batch_config = self.datasource.add_batch_config(batch_config)
-            batch_config.set_data_asset(self)
-
         batch_config.set_data_asset(self)
         self.batch_configs.append(batch_config)
+        if self.datasource.data_context:
+            try:
+                batch_config = self.datasource.add_batch_config(batch_config)
+            except DataContextError:
+                self.batch_configs.remove(batch_config)
+                raise
+        else:
+            self.batch_configs.append(batch_config)
```

Comment:
Does this add it twice because of line 292?

Reply:
Yeah looks like it did. Removed this line in https://github.com/great-expectations/great_expectations/pull/9201/files

Reply:
I hadn't noticed you did that. Thanks!